src/common/interval.h

```diff
@@ -18,6 +18,10 @@ class interval {
     interval(const interval& a, const interval& b)
       : min(fmin(a.min, b.min)), max(fmax(a.max, b.max)) {}
 
+    bool is_empty() const {
+        return max < min;
```

Comment:
`is_empty()` and `interval::empty` are slightly different things. I don't think it'll create a problem, but my engineer brain pings when I see stuff like this.

Reply:
You mean the value versus the property? The is_empty property is true if the interval contains no points. Thus, there are infinitely many ways to construct an interval such that it contains no points, yet it's quite useful (and efficient) to have a canonically empty interval.

Reply:
yeah, exactly.

Reply:
Ha. I think I needed `aabb::is_empty()` and then refactored so I didn't need it after all. And thus I don't need `interval::is_empty()`. Problem solved for now.